sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -78,6 +79,26 @@ final class DataFrameWriter private[sql](df: DataFrame) {
   }
 
   /**
+   * Set the trigger interval for the stream query.
+   *
+   * @since 2.0.0
+   */
+  def trigger(interval: Duration): DataFrameWriter = {
+    this.extraOptions += ("triggerInterval" -> interval.toMillis.toString)
```

Comment:
I think we probably should create another var, rather than converting this back/forth from a string.  Also, how would we extend this to support other types of triggers?  The design doc suggests using using classes I think.